.github/workflows/trivy.yml

```diff
@@ -0,0 +1,62 @@
+name: Precommit - Scan security vulnerabilities
+on:
+  pull_request:
+    branches:
+      - '*'
+    paths-ignore:
+      - 'docs/**'
+      - 'README.md'
+      - 'CHANGELOG.md'
+      - 'PROJECT'
+      - 'LICENSE'
+      - 'mesh-worker-service/README.md'
+      - 'tools/README.md'
+jobs:
+  build:
+    name: Scan
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Set up GO 1.18
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.18
+        id: go
+
+      - name: InstallKubebuilder
+        run: |
+          curl -L https://github.com/kubernetes-sigs/kubebuilder/releases/download/v2.3.1/kubebuilder_2.3.1_linux_amd64.tar.gz | tar -xz -C /tmp/
+          sudo mv /tmp/kubebuilder_2.3.1_linux_amd64 /usr/local/kubebuilder
+          export PATH=$PATH:/usr/local/kubebuilder/bin
+
+      - name: Install operator-sdk
+        run: |
+          RELEASE_VERSION=v1.14.0
+          curl -LO "https://github.com/operator-framework/operator-sdk/releases/download/${RELEASE_VERSION}/operator-sdk_linux_amd64"
+          chmod +x operator-sdk_linux_amd64 && sudo mkdir -p /usr/local/bin/ && sudo mv operator-sdk_linux_amd64 /usr/local/bin/operator-sdk
+
+      - name: Set up yq
+        run: |
+          sudo wget https://github.com/mikefarah/yq/releases/download/v4.6.0/yq_linux_amd64 -O /usr/bin/yq
+          sudo chmod +x /usr/bin/yq
+          yq --help
+
+      - name: Build operator image
+        run: |
+          make generate
+          make helm-crds
+          image="function-mesh-operator:latest"
+          IMG=${image} make docker-build-skip-test
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: 'function-mesh-operator:latest'
```

Comment:
one thing that may need to the discussion is the severity, the default value is `UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL`, do we need to ignore some of them, like `UNKNOWN`, `LOW` severity vulnerabilities?

@streamnative/serverless @hpvd

Reply:
great to see taking action on this!

would vote for `suppress nothing`. It mostly isn't much work to update a dependency for the benefit of better security.
If there are vulnerabilities with no fix available yet, one could suppress them until a fix is available...
(but one have to take care to not suppress them forever)

One could also think one step further, and not only update dependencies with known security flaws, but directly always when a new version of a dependency is released.

This could be comfortably done, using dependabot and let this tool automatically start new pull request for update
see: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuring-dependabot-version-updates

With this, trivy will have much less findings and one is one step ahead of many security findings...

Reply:
make sense to me